ledfx/effects/digitalrain2d.py

```diff
@@ -0,0 +1,204 @@
+import logging
+import random
+
+import numpy as np
+import voluptuous as vol
+
+from ledfx.color import validate_gradient
+from ledfx.effects.gradient import GradientEffect
+from ledfx.effects.twod import Twod
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class Line:
+    def __init__(self, nx, color, offset, speed):
+        self.nx = nx
+        self.ny = 0
+        self.color = color
+        # per line modifyer on speed, norm 1.0
+        self.speed = speed
+        self.tail = 0.5
+        self.offset = offset
+        self.impulse_index = int(offset * 3)
+
+    def update(self, run_seconds, passed, tail, impulse):
+        # calculate how much to move
+        movement = passed / run_seconds * (1 + impulse[self.impulse_index])
+        self.tail = tail
+        # adjust for the code lines own speed
+        self.ny += movement * self.speed
+        if self.ny > (1 + self.tail):
+            return False
+        return True
+
+    def draw(self, draw, image, width, beat_osc):
```

Comment:
I think this is the first time we're not using the effects draw method?

Probably outline in the docstring how and why this is here as opposed to in the effect.

We probably want a reeeeealy good reason for having draw here as opposed to the parent class.

Reply:
It's just a draw method inside a object that gets called from other draw. I can rename it, but not seeing the concern

Reply:
So its draw methods all the way down, these seems a standard paradigm to me